### PR TITLE
Fix v2 assets validation

### PR DESCRIPTION
**Pull request by maxep**
### What and why?

- Fix xcframework validation of v2 modules.
- Fix `DatadgSDK.podspec`

### Review checklist
- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [ ] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [ ] Run unit tests
- [ ] Run integration tests
- [x] Run smoke tests

### CODE REVIEW

File: tools/distribution/src/release/assets/gh_asset.py [hunk 1]
```diff
@@ -203,27 +205,72 @@ def validate(self, zip_directory: DirectoryMatcher, in_version: Version) -> bool
             return False
             
         zip_directory.get('Kronos.xcframework').assert_it_has_files([
-            'ios-arm64_armv7',
-            'ios-arm64_armv7/dSYMs/*.dSYM',
-            'ios-arm64_armv7/**/*.swiftinterface',
+            'ios-arm64_arm64e',
+                'ios-arm64_arm64e/dSYMs/*.dSYM',
+                'ios-arm64_arm64e/**/*.swiftinterface',
```
Comment (ncreated) [inline tools/distribution/src/release/assets/gh_asset.py:210]:
```suggestion
            'ios-arm64_arm64e/dSYMs/*.dSYM',
            'ios-arm64_arm64e/**/*.swiftinterface',
```

File: DatadogSDK.podspec [hunk 2]
```diff
@@ -21,12 +21,11 @@ Pod::Spec.new do |s|
...
 
   s.deprecated_in_favor_of =
-    'DatadogCore, DatadogLogs, DatadogRUM, DatadogTrace, and DatadogWebViewTracking'
+    'DatadogCore, DatadogLogs, DatadogRUM, and DatadogTrace'
 
   s.dependency 'DatadogCore', s.version.to_s
   s.dependency 'DatadogLogs', s.version.to_s
   s.dependency 'DatadogRUM', s.version.to_s
   s.dependency 'DatadogTrace', s.version.to_s
-  s.dependency 'DatadogWebViewTracking', s.version.to_s
```
Comment (ncreated) [inline DatadogSDK.podspec:30]: question/ What's the explanation of this change 🤔? `DDWVT.podspec` doesn't support `tvOS` hence it failed on `tvOS` compatibility check for `DatadogSDK.podspec`?

File: tools/distribution/src/release/assets/gh_asset.py [hunk 3]
```diff
@@ -203,27 +205,72 @@ def validate(self, zip_directory: DirectoryMatcher, in_version: Version) -> bool
...
 
             'ios-arm64_i386_x86_64-simulator',
             'ios-arm64_i386_x86_64-simulator/dSYMs/*.dSYM',
             'ios-arm64_i386_x86_64-simulator/**/*.swiftinterface',
         ])
 
         return True
+    
+class DatadogModuleXCFrameworkValidator(XCFrameworkValidator):
```
Comment (ncreated) [inline tools/distribution/src/release/assets/gh_asset.py:219]: Nice 👌

**Review by ncreated — APPROVED** [review]
Looks good 👍. I spawned [test job](https://app.bitrise.io/build/6ffd907a-d41d-4849-ab85-3827c119d775?tab=log) for `2.0.0-beta1` with `DD_RELEASE_DRY_RUN=1`, let's see if it will work.

**Review by ncreated — APPROVED** [review]
🏅 
```
   → Validating each `XCFramework`:
       → Datadog.xcframework - SKIPPING for 2.0-beta.1
       → Kronos.xcframework - SKIPPING for 2.0-beta.1
       → DatadogInternal.xcframework - OK
       → DatadogCore.xcframework - OK
       → DatadogLogs.xcframework - OK
       → DatadogTrace.xcframework - OK
       → DatadogRUM.xcframework - OK
       → DatadogWebViewTracking.xcframework - OK
       → DatadogObjc.xcframework - OK
       → DatadogCrashReporting.xcframework - OK
       → CrashReporter.xcframework - OK
   → the content of `.zip` archive is correct
```

File: DatadogSDK.podspec (hunk 2 above)
Comment (maxep) [reply DatadogSDK.podspec:30]: Exactly! here the checks: 

```
[!] DatadogSDK has been deprecated in favor of DatadogCore, DatadogLogs, DatadogRUM, DatadogTrace, and DatadogWebViewTracking
 -> DatadogSDK (2.0.0-beta1)
    - NOTE  | url: The URL (https://twitter.com/datadoghq) is not reachable.
    - NOTE  | xcodebuild:  note: Using codesigning identity override: -
    - NOTE  | [iOS] xcodebuild:  note: Building targets in dependency order
    - NOTE  | [iOS] xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'DatadogInternal' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'DatadogWebViewTracking' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'DatadogTrace' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'DatadogLogs' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'DatadogCore' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'DatadogRUM' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  note: Metadata extraction skipped. No AppIntents.framework dependency found. (in target 'App' from project 'App')
    - ERROR | [tvOS] unknown: Encountered an unknown error (The platform of the target `App` (tvOS 11.0) is not compatible with `DatadogWebViewTracking (2.0.0-beta1)`, which does not support `tvOS`.) during validation.
Analyzed 1 podspec.
```

**Pull request merged**